### PR TITLE
Do not const_cast/modify sh.mem. objetcs from upstream procsesses

### PR DESCRIPTION
**Pull request by shahor02**
The tracks produced in the tracks and supposedly stored in the AOD as is were modified but the QC tasks after const_casting objects in the shared memory. It looks like only standalone TPC tracks were affected (were propagated to the DCA) so the damage was limited.

### CODE REVIEW

**Comment by Barthelemy** [comment]
we have a release tomorrow. Is it urgent or do we wait for the approval of the detectors ?

**Comment by shahor02** [comment]
Hi @Barthelemy , I've tested it locally. I would not wait for validation by detectors since this is a fix to respect basic aspect of O2 processing: the DPL input must treated as const.

**Review by Barthelemy — APPROVED** [review]
ok, then I approve and I will merge

**Pull request merged**
